packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -283,6 +285,20 @@ class ResidentWebRunner extends ResidentRunner {
         ).send();
       }
     }
+    if (device is ChromeDevice && debuggingOptions.browserLaunch) {
+      try {
+        final Chrome chrome = await ChromeLauncher.connectedInstance;
+        final ChromeTab chromeTab = await chrome.chromeConnection.getTab((ChromeTab chromeTab) {
+          return chromeTab.url.contains(debuggingOptions.hostname);
```

Comment:
Should we use some sort of nonce in case there are multiple tabs?

Reply:
We don't really support that in the launched browser case